parser/util/Environment.js

```diff
@@ -0,0 +1,35 @@
+import { LinkedList } from './LinkedList';
+
+/**
+ * A special key-value pair linked list that can lookup values of a certain key.
+ */
+export class Environment extends LinkedList {
+  constructor (head, tail) {
+    super(head, tail);
+    let hed = this.head;
```

Comment:
Use `const`

Also, please use a different name. The name `hed` is not very descriptive.